c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -157,21 +157,21 @@ public:
     auto req = KJ_REQUIRE_NONNULL(out, "already called disconnect()").sendDataRequest(
         MessageSize { 8 + message.size() / sizeof(word), 0 });
     req.setData(message);
-    return req.send();
+    return req.send().then([this, size = message.size()](){ sentBytes += size; });
```

Comment:
I think you can count the `sentBytes` before actually calling `send()`, to avoid the need to use a `.then()`.

Technically, the promise returned by `send()` doesn't really tell you when the bytes have "been sent" anyway. It resolves as soon as it's OK to send more. The bytes may still be in a buffer somewhere, and any number of errors could still prevent the bytes from ever being delivered.

Reply:
So I suppose there's no way to exclude the bytes that were not delivered without modifying the underlying `StreamingRequest` to do some more accounting. Do you think it's worth the effort here? Or maybe we actually prefer to count the bytes that we attempted to send.

Reply:
I mean, theoretically speaking, if you _really_ want to track what was actually delivered, you'd have to query the operating system's TCP layer to find out which packets were ACKed. And that assumes there's no proxies between you and the final destination; if there are, then you need to ask those proxies to report what they managed to deliver, etc...

I don't think any of this is worth it. If we tried to send the thing, then count it. I don't think the error rate will be significant here.